dkc/core/rest/utils.py

```diff
@@ -0,0 +1,19 @@
+from django.core.exceptions import ValidationError as DjangoValidationError
+from rest_framework import serializers
+
+
+class FullCleanModelSerializer(serializers.ModelSerializer):
+    def validate(self, data):
+        if self.partial:  # partial update
+            for k, v in data.items():
+                setattr(self.instance, k, v)
```

Comment:
We're mutating the only copy of `self.instance` here, which means that validation (even if unsuccessful) has side effects on the existing model. If a request doesn't terminate immediately after this, we'll end up with an invalid model, which could even be a problem for debugging.

Reply:
> If a request doesn't terminate immediately after this, we'll end up with an invalid model, which could even be a problem for debugging.

Do you mean to say that a model instance object may be shared across different requests...?

Reply:
Ignore deleted comment, I misread your last sentence.